pkg/azure/provider.go

```diff
@@ -606,3 +619,16 @@ func getCurve(crv kv.JSONWebKeyCurveName) (elliptic.Curve, error) {
 		return nil, fmt.Errorf("curve %s is not suppported", crv)
 	}
 }
+
+func parsePrivateKey(block []byte) (interface{}, error) {
```

Comment:
Can you PTAL at https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/pkg/secrets-store/utils.go#L773-L791 and make sure the conversion of these types match?

Reply:
The `MarshalPKCS8PrivateKey` is done to ensure backward compatibility and also compatibility with the driver conversion while creating K8s secrets.